src/static/js/utils/representation.js

```diff
@@ -37,18 +38,25 @@ ObjectAddChange.prototype.applyOnText = function(text){
   return text.insert(this._content, this._pos);
 };
 ObjectAddChange.prototype.applyOnPos = function(pos){
-  return pos + (this._pos <= pos ? this.size() : 0);
+  if(!this._author){
+    return pos + (this._pos <= pos ? this.size() : 0);
+  } else {
+    return pos;
+  }
```

Comment:
Could we change the parameter "pos" when needed and return it at the end ?
This would make only one return.